sdk/keyvault/azure-security-keyvault/azure/security/keyvault/vault_client.py

```diff
@@ -3,15 +3,31 @@
 # Licensed under the MIT License. See LICENSE.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import
+    from azure.core import Configuration
+    from azure.core.credentials import SupportsGetToken
+    from azure.core.pipeline.transport import HttpTransport
+    from typing import Any, Mapping, Optional
+
+from ._internal import _KeyVaultClientBase
 from .keys._client import KeyClient
 from .secrets._client import SecretClient
 
 
-class VaultClient(object):
-    def __init__(self, vault_url, credentials, config=None, **kwargs):
-        self._vault_url = vault_url.strip(" /")
-        self._secrets = SecretClient(self._vault_url, credentials, config=config, **kwargs)
-        self._keys = KeyClient(self._vault_url, credentials, config=config, **kwargs)
+class VaultClient(_KeyVaultClientBase):
+    def __init__(self, vault_url, credential, config=None, transport=None, api_version=None, **kwargs):
+        # type: (str, SupportsGetToken, Configuration, Optional[HttpTransport], Optional[str], **Any) -> None
```

Comment:
>SupportsGetToken [](start = 22, length = 16)

This doesn't jive with the changes from your other PR, just a heads up depending on the order you merge them.